creator/ingest_runs/signals.py

```diff
@@ -13,3 +16,41 @@ def ingest_run_pre_save(sender, instance, *args, **kwargs):
     if instance.versions.count() > 0:
         instance.name = instance.compute_name()
         instance.input_hash = instance.compute_input_hash()
+
+
+@receiver(post_transition, sender=IngestRun)
+def ingest_run_post_transition(sender, instance, name, source, target,
+                               *args, **kwargs):
+    """
+    Create a corresponding Event after the status of an IngestRun is changed.
+    """
+    if target == "running":
+        target = "started"
+    TARGET_EVENT_TYPES = {
+        "started": "IR_STA",
+        "complete": "IR_COM",
+        "canceled": "IR_CAN",
+        "failed": "IR_FAI",
+    }
+    versions = list(instance.versions.all())
+    username = instance.creator.display_name or "Anonymous user"
+    if target in {"complete", "failed"}:
+        message = (
+            f"IngestRun {instance.id} {target} for file versions {versions} "
+        )
+    else:
+        message = (
+            f"{username} {target} IngestRun {instance.id} for file versions "
+            f"{versions} "
+        )
+    
+    first_version = instance.versions.first()
+    ev = Event(
+        study=first_version.study,
+        file=first_version.root_file,
+        version=first_version,
+        user=instance.creator,
```

Comment:
You may want to consider excluding the user if the event is one that is not created in response to a request from the user.